federation/planner.go

```diff
@@ -145,22 +192,34 @@ func (e *Planner) planObject(typ *graphql.Object, selectionSet *graphql.Selectio
 			return nil, fmt.Errorf("typ %s has no field %s", typ.Name, selection.Name)
 		}
 
-		fieldInfo := e.schema.Fields[field]
-
-		// Prioritize resolving as many fields as we can in the current service
-		if fieldInfo.Services[service] {
-			localSelections = append(localSelections, selection)
-		} else {
-			serviceWithField := ""
+		var hasCustomService bool
+		hasCustomService, localSelections, selectionsByService = e.maybeCustomService(
+			typ.Name,
+			service,
+			selection,
+			localSelections,
+			selectionsByService)
+
+		if !hasCustomService {
+			fieldInfo := e.schema.Fields[field]
+			// Prioritize resolving as many fields as we can in the current service
+			if fieldInfo.Services[service] {
+				localSelections = append(localSelections, selection)
+			} else {
+				if len(fieldInfo.Services) != 1 {
```

Comment:
I think we want this check outside of the else statement. Even if we're prioritizing adding it to the current service, we still want the check of whether or not the field is on multiple services and doesn't have a service selector

Reply:
We talked offline, move this check out of the branch.